community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/CsvInput.java

```diff
@@ -237,4 +251,77 @@ public Collector badCollector()
     {
         return badCollector;
     }
+
+    @Override
+    public Estimates calculateEstimates( ToIntFunction<Value[]> valueSizeCalculator ) throws IOException
+    {
+        long[] nodeSample = sample( nodeDataFactory, nodeHeaderFactory,
+                ( header, data ) -> new InputNodeDeserialization( header, data, groups, idType.idsAreExternal() ), valueSizeCalculator,
+                node -> node.labels().length );
+        long[] relationshipSample = sample( relationshipDataFactory, relationshipHeaderFactory,
+                ( header, data ) -> new InputRelationshipDeserialization( header, data, groups ), valueSizeCalculator, entity -> 0 );
+        return knownEstimates(
+                nodeSample[0], relationshipSample[0],
+                nodeSample[1], relationshipSample[1],
+                nodeSample[2], relationshipSample[2],
+                nodeSample[3] );
+    }
+
+    private <E extends InputEntity> long[] sample( Iterable<DataFactory<E>> dataFactories,
+            Header.Factory headerFactory, BiFunction<Header,CharSeeker,InputEntityDeserialization<E>> deserialization,
+            ToIntFunction<Value[]> valueSizeCalculator, ToIntFunction<E> additionalCalculator ) throws IOException
+    {
+        long[] estimates = new long[4]; // [entity count, property count, property size, labels (for nodes only)]
+        for ( DataFactory<E> dataFactory : dataFactories ) // one input group
+        {
+            // One group of input files
+            Header header = null;
+            RawIterator<CharReadable,IOException> dataItems = dataFactory.create( config ).stream();
+            while ( dataItems.hasNext() )
+            {
+                CharReadable stream = dataItems.next();
+                // TODO for now only a single 1MB chunk from the start of each file is sampled.
```

Comment:
this todo should be fixed or converted into the card

Reply:
I think it's fine as it is... will remove it.